datafusion/core/src/execution/session_state.rs

```diff
@@ -1887,8 +1898,8 @@ impl OptimizerConfig for SessionState {
         &self.execution_props.alias_generator
     }
 
-    fn options(&self) -> &ConfigOptions {
-        self.config_options()
+    fn options(&self) -> Arc<ConfigOptions> {
```

Comment:
If we are going to change this I suggest changing to `&Arc<ConfigOptions>`, that way the caller can choose if they want to clone the Arc or not, and normal borrowing rules work nicely